src/main/java/ca/on/oicr/pineryreports/util/SampleUtils.java

```diff
@@ -129,6 +134,28 @@ private SampleUtils() {
     return dto -> getAttribute(ATTR_RECEIVE_DATE, dto) == null;
   }
 
+  public static Predicate<SampleDto> byDnaLibrary() {
+    return byRnaLibrary().negate();
+  }
+
+  public static Predicate<SampleDto> byRnaLibrary() {
+    return dto -> {
+      String designCode = getAttribute(ATTR_SOURCE_TEMPLATE_TYPE, dto);
+      if (designCode == null)
+        throw new IllegalArgumentException("Library is missing a library design code; is " + dto.getId() + " really a library?");
+      return RNA_LIBRARY_DESIGN_CODES.contains(designCode);
+    };
+  }
+
+  public static boolean isNonIlluminaLibrary(SampleDto dilution) {
+    if (dilution.getSampleType() == null) throw new IllegalArgumentException("Dilution " + dilution.getName() + " has no sample_type");
+    return !dilution.getSampleType().contains("Illumina"); // note the negation here
+  }
+
+  public static Predicate<SampleDto> byNonIlluminaLibrary() {
+    return dto -> isNonIlluminaLibrary(dto);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)